src/MongoDB.Driver/Encryption/MongocryptdFactory.cs

```diff
@@ -0,0 +1,182 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace MongoDB.Driver.Encryption
+{
+    internal static class EncryptionExtraOptionsValidator
+    {
+        #region static
+        private static readonly Dictionary<string, Type[]> __supportedExtraOptionKeys = new Dictionary<string, Type[]>
```

Comment:
This field is more than just keys... rename to:

    __supportedExtraOptions

?